templates/components/vectordbs/python/none/generate.py

```diff
@@ -15,16 +20,62 @@
 logger = logging.getLogger()
 
 
+def get_vector_store():
+    if not os.path.exists(STORAGE_DIR):
+        vector_store = SimpleVectorStore()
+        return vector_store
+    else:
+        return SimpleVectorStore.from_persist_dir(STORAGE_DIR, namespace="default")
+
+
+def get_doc_store():
+    if not os.path.exists(STORAGE_DIR):
+        docstore = SimpleDocumentStore()
+        return docstore
+    else:
+        return SimpleDocumentStore.from_persist_dir(STORAGE_DIR)
+
+
 def generate_datasource():
     init_settings()
     logger.info("Creating new index")
+
     # load the documents and create the index
     documents = get_documents()
-    index = VectorStoreIndex.from_documents(
-        documents,
+    docstore = get_doc_store()
+    vector_store = get_vector_store()
+    storage_context = StorageContext.from_defaults(
+        vector_store=vector_store,
+        docstore=docstore,
+    )
+
+    # Create ingestion pipeline
+    ingestion_pipeline = IngestionPipeline(
+        transformations=[
+            SentenceSplitter(
+                chunk_size=Settings.chunk_size,
+                chunk_overlap=Settings.chunk_overlap,
+            ),
+            Settings.embed_model,
+        ],
+        docstore=docstore,
+        docstore_strategy="upserts_and_delete",
+    )
+
+    # llama_index having an typing issue when passing vector_store to IngestionPipeline
+    # so we need to set it manually after initialization
+    ingestion_pipeline.vector_store = vector_store
+
+    # Run the ingestion pipeline and store the results
+    nodes = ingestion_pipeline.run(show_progress=True, documents=documents)
+
+    # Create the index
```

Comment:
Should be possible without creating the index and without storage context

Reply:
The storage context is redundant in this case, but we still need the index because of the retrieval